tracer/src/Datadog.InstrumentedAssemblyGenerator/ProfilerMetadataImporter.cs

```diff
@@ -98,8 +99,11 @@ private void ImportInstrumentedAssemblyMetadata()
 
                     case MetadataTable.Field:
                     {
-                        var typeToAddField = OriginalModule.Types.SingleOrDefault(t =>
-                                                                                      t.Name.String.Equals(tokensAndName.Value.Type, StringComparison.InvariantCultureIgnoreCase));
+                        // TODO: Add parent type (declaring type) to be able to identify two nested types with the same name (internal ticket id 1029) -- see also MetadataNameParser.Parse
+                        // we choose the first one just to make it work.
+                        //var typeToAddField = OriginalModule.GetTypes().SingleOrDefault(t =>
```

Comment:
```suggestion
```

Reply:
Fixed in 5a103fbe57cedbc9e13568d4372c82a921c46e26